tests/data_context/store/test_datasource_store.py

```diff
@@ -164,3 +175,83 @@ def test_datasource_store_with_inline_store_backend(
 
     assert isinstance(res, DatasourceConfig)
     assert res.to_json_dict() == datasource_config.to_json_dict()
+
+
+def test_datasource_store_set_by_name(
+    empty_datasource_store: DatasourceStore,
+    datasource_config: DatasourceConfig,
+    datasource_name: str,
+) -> None:
+    assert len(empty_datasource_store.list_keys()) == 0
+
+    empty_datasource_store.set_by_name(
+        datasource_name=datasource_name, datasource_config=datasource_config
+    )
+
+    assert len(empty_datasource_store.list_keys()) == 1
+
+
+def test_datasource_store_retrieve_by_name(
+    datasource_name: str,
+    datasource_config: DatasourceConfig,
+    datasource_store_with_single_datasource: DatasourceStore,
+) -> None:
+    actual_config: DatasourceConfig = (
+        datasource_store_with_single_datasource.retrieve_by_name(
+            datasource_name=datasource_name
+        )
+    )
+    assert datasource_config.to_dict() == actual_config.to_dict()
+
+
+def test_datasource_store_delete_by_name(
+    datasource_name: str,
+    datasource_store_with_single_datasource: DatasourceStore,
+) -> None:
+    assert len(datasource_store_with_single_datasource.list_keys()) == 1
+
+    datasource_store_with_single_datasource.delete_by_name(
+        datasource_name=datasource_name
+    )
+
+    assert len(datasource_store_with_single_datasource.list_keys()) == 0
+
+
+def test_datasource_store_update_by_name(
+    datasource_name: str,
+    datasource_config: DatasourceConfig,
+    datasource_store_with_single_datasource: DatasourceStore,
+) -> None:
+    updated_base_directory: str = "foo/bar/baz"
+
+    updated_datasource_config = copy.deepcopy(datasource_config)
+    updated_datasource_config.data_connectors["tripdata_monthly_configured"][
+        "base_directory"
+    ] = updated_base_directory
+
+    datasource_store_with_single_datasource.update_by_name(
+        datasource_name=datasource_name, datasource_config=updated_datasource_config
+    )
+
+    key: DataContextVariableKey = DataContextVariableKey(
```

Comment:
My preference would be to remove the annotations on `key` and `actual_config` since you instantiate/cast it in the same line and `mypy` will infer this.